substrate-ros/player_codegen/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright 2018-2019 Airalab <research@aira.life>
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+//! This module provides macro for build message players and join it in one future
+
+#![feature(async_await)]
+#![feature(async_closure)]
+#![feature(core_intrinsics)]
+#![feature(trace_macros)]
+#![feature(proc_macro_quote)]
+
+extern crate proc_macro;
```

Comment:
Not needed in edition2018

Reply:
```extern crate proc_macro``` still needed in edition 2018 for proc-macro modules
https://doc.rust-lang.org/reference/procedural-macros.html#function-like-procedural-macros
```error[E0432]: unresolved import 'proc_macro'``` occur if not specified ```extern crate proc_macro```